### DISCUSSION TITLE
form values are invalid when rtk-query is used

### DISCUSSION THREAD

**Discussion by pavelspichonak** [react-hook-form/react-hook-form#9882] [Q&A]
I'm using:
@hookform/resolvers 2.9.10
react 18.2.0
react-dom 18.2.0
react-hook-form 7.43.0
yup 0.32.11

[here is sandbox](https://codesandbox.io/s/react-hook-form-v7-with-yup-forked-8m12uk?file=/src/App.js)

`shouldUnregister: true` and `values` are used in `useForm`.

In this sandbox all works good. Form get `defaultValues` and `values` and on submit we get form values and have no **yup** validation errors. We can change form values using `add new values` button changing value of `values` property in `useForm` hook. All is good, all works.

I do not know why, but when on the same page where `useForm` hook is used, we use some `query` using `rtk-query` library like [in this example](https://redux-toolkit.js.org/rtk-query/usage/queries#query-hook-usage-example)
for example like this:

```
  const {
    data: post,
  } = useGetPostQuery(id)

  return (
    <div>
      <MyForm /> // MyForm component uses useForm inside 
    </div>
  ) 

``` 

on the screen we see that form inputs, selects, checkboxes have values which we passed to `useForm` using `values` property, but when we submit form we have `yup` validation errors and it looks like `useForm` passes `values` to inputs, selects, checkboxes but on submit `useForm` passes `defaultValues` to `yup` to validate.

In other words when we use query from `rkq-query`  it looks like `yup` get `defaultValues` instead of `values`.

And one more thing. In form we do not use data which we get from `query`, we use this query to get data for other place on the page.

If we remove `query` which requests some data, then `useForm` and submit works great without any `yup` errors. Very strange behaviour and I have no idea how requesting data using `query` can affect `useForm` hook. Moreover data from `query` is not used in `useForm` at all...

Does anybody have any idea? Thanks in advance.

cc @bluebill1049

**Comment by bluebill1049** [comment]
hey @pavelspichonak any chance you could try to invoke `unregister` manually instead of using `shouldUnregister: true`? See if the issue still persist

**Reply by pavelspichonak** [reply]
Issue happens before form or its inputs, selects and checkboxes are unmounted. Without `shouldUnregister: true` it works good.

**Reply by bluebill1049** [reply]
`shouldUnregister: true` will unmount input and mutate both input reference, `values` and also `defaultValues`, if that's causing you the issue, is it possible to skip that config?

**Reply by pavelspichonak** [reply]
Also I tried to add `shouldUnregister: true` to `useController` and it still works good.

**Reply by bluebill1049** [reply]
Yes, 

one fire at the component `useEffect` level
the other fire at `useForm` `useEffect` level

that's the difference.

**Reply by pavelspichonak** [reply]
I'll try to think tomorrow. Thanks.
